### PR TITLE
Pr/2904 document config changes

### PR DESCRIPTION
**Pull request by b-abel**
This documents the changes made for #2904.

### CODE REVIEW

File: site/documentation/content/admin-guide/hono-kafka-client-configuration.md [hunk 1]
```diff
@@ -47,110 +54,83 @@ type configuration is ignored.
...
-in order to implement the message delivery semantics defined by Hono's Telemetry and Event APIs.
+The following properties can _not_ be set because
+`org.eclipse.hono.client.kafka.producer.MessagingKafkaProducerConfigProperties` uses fixed values instead in order to
+implement the message delivery semantics defined by Hono's Telemetry and Event APIs.
 
 | Property Name     | Fixed Value |
 | :---------------- | :---------- |
 | `key.serializer`    | `org.apache.kafka.common.serialization.StringSerializer` |
 | `value.serializer`   | `io.vertx.kafka.client.serialization.BufferSerializer` |
 | `enable.idempotence` | `true` |
```
Comment (calohmn) [inline site/documentation/content/admin-guide/hono-kafka-client-configuration.md:76]: How about adding a sentence here that a configured `client.id` value will only be adopted as a prefix, with the client name and a unique identifier getting attached to it? That should then be copied to the producer and admin config sections.

File: site/homepage/content/release-notes.md [hunk 2]
```diff
@@ -13,6 +13,10 @@ description = "Information about changes in recent Hono releases. Includes new f
 * The JDBC DB based registry now also supports enforcement of registration limits configured at the tenant level.
   In particular, the maximum number of devices and the maximum number of credentials per device can be set in
   a tenant's registration-limits property. Please refer to the User Guide for details.
+* Kafka clients used by a component can now be configured individually instead of all clients of a type using the same
+  configuration. The prefixes that are prepended to the configuration properties of the native Kafka client have
+  changed. Existing configurations have to be adapted. Please refer to
```
Comment (calohmn) [inline site/homepage/content/release-notes.md:18]: How about:
> Existing configurations using `producerConfig`, `consumerConfig` and/or `adminClientConfig` configuration properties have to be adapted.

or

> For existing configurations using just `commonClientConfig` properties, no change is needed. Other configurations with specific consumer/producer/admin client properties have to be adapted.

?

File: site/documentation/content/admin-guide/hono-kafka-client-configuration.md (hunk 1 above)
Comment (b-abel) [reply site/documentation/content/admin-guide/hono-kafka-client-configuration.md:77]: Please see my changes here.

File: site/documentation/content/admin-guide/hono-kafka-client-configuration.md [hunk 3]
```diff
@@ -50,108 +58,89 @@ Kafka based APIs. The producers created by the factory are configured with insta
...
-The factory can be configured to use TLS for authenticating the brokers in the Kafka cluster during connection
-establishment and optionally for authenticating to the broker using a client certificate.
-To use this, a Kafka Producer configuration as described in
-[Kafka documentation - section "Security"](https://kafka.apache.org/documentation/#security_configclients) needs to be
-provided. The properties must be prefixed with `HONO_KAFKA_PRODUCERCONFIG_` and `hono.kafka.producerConfig.` respectively as
-shown in [Producer Configuration Properties]({{< relref "#producer-configuration-properties" >}}).
-The complete reference of available properties and the possible values is available in 
-[Kafka documentation - section "Producer Configs"](https://kafka.apache.org/documentation/#producerconfigs).
+If the property `client.id` is provided, its value is appended by a generated unique suffix that provides additional
+information.
```
Comment (calohmn) [inline site/documentation/content/admin-guide/hono-kafka-client-configuration.md:79]: To also cover the "if it's not provided" case, I would suggest using something like this instead:
> Kafka clients used in Hono will get a unique client identifier, containing client name and component identifier. If the property `client.id` is provided, its value will be used as prefix for the created client identifier.

Comment (b-abel) [reply site/documentation/content/admin-guide/hono-kafka-client-configuration.md]: Done.

**Review by calohmn — APPROVED** [review]
LGTM

**Pull request merged**